### PR TITLE
feat: peek overlay nested properties + perf improvements

### LINKED ISSUE #23054: [Task]: Remove usage of entity markers for peek overlay (perf improvement)
### Description
Reduce the amount of time taken when editing code in JsEditor.
In some cases, we’ve seen the application slow down because of the way we handled text marking on hover.
https://github.com/appsmithorg/appsmith/issues/21467

So we're moving away from text marking to AST based peek overlay.

---

- [x] Handle AST parsing for JsObject and property pane separately.
- [x] Handle False hovers (hover highlight and ast parsing)
  - [x] Handle JsKeywords (export default, return).
  - [x] Filter object property definitions, local variables and function calls.
- [ ] Handle cypress failures (if any)

### LINKED ISSUE #23057: [Task]: Extend peek overlay with nested properties. (e.g. `Table1.data[0].id`)
- [x] Filter json data from evaluation based on entityDefinitions
- [x] Identify hardcoded array & string accessors.

### PR DESCRIPTION
**Pull request by eco-monk**
Fixes #23057
Fixes #23054

## Description
TL;DR Added support for peeking on nested properties. e.g. `Api1.data[0].id`.

This won't work when:
-  local variables are involved in the expression. 
e.g. `Api1.data[x].id` won't support peeking at the variable `[x]` or anything after that.
- library code is involved e.g. `moment`, `_` etc...
- when functions are called. e.g. Api1.data[0].id.toFixed()

Because these cases requires evaluation.

image

#### Media (old vs new)
https://www.loom.com/share/dedcf113439c4ee2a19028acca54045e

## Performance improvements:
- Use AST to identify expressions instead marking text manually.
- This reduces the number of markers we process (~ half).

- Before
image

- After
image

- AST logs
https://www.loom.com/share/ddde93233cc8470ea04309d8a8332240

#### Type of change
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?
- [x] Manual
- [x] Jest
- [x] Cypress
>
>
#### Test Plan
https://github.com/appsmithorg/TestSmith/issues/2402

#### Issues raised during DP testing
https://github.com/appsmithorg/appsmith/pull/23414#issuecomment-1553164908

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [x] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [x] Cypress test cases have been added and approved by SDET/manual QA
- [x] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by ramsaptami** [comment]
- [ ] Incorrectly shows peek overlay
image
image
image

- [ ] overlay does not show up on key-value format 
image

- [ ] Unsure if we support `this.`
image

File: app/client/src/components/editorComponents/CodeEditor/index.tsx [hunk 1]
```diff
@@ -649,35 +652,54 @@ class CodeEditor extends Component<Props, State> {
...
+              tokenElement.classList.contains("cm-property") ||
+              // array indices - [0]
+              tokenElement.classList.contains("cm-number") ||
+              // string accessor - ["x"]
+              tokenElement.classList.contains("cm-string")
+            ) {
+              this.showPeekOverlay(lineExpression, paths, tokenElement);
+            } else if (tokenElement.classList.contains("cm-keyword")) {
+              // this keyword for jsObjects
+              if (this.props.isJSObject && tokenElement.innerHTML === "this") {
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/index.tsx:692]: Too many conditions here. Can we extract it into a helper?

Comment (eco-monk) [reply app/client/src/components/editorComponents/CodeEditor/index.tsx]: Done

**Comment by ramsaptami** [comment]
LGTM
cc: @eco-monk

**Review by hetunandu — APPROVED** [review]
